### PR TITLE
6 remove not necessary

### LINKED ISSUE #6: Remove all stuff that is not absolutely necessary right now
Remove all stuff that is not absolutely necessary right now. We decided that we don't understand or need many things that were copied from [cpp-best-practices/cmake-template](https://github.com/cpp-best-practices/cmake_template) and it is better to remove all of that, and gradually add stuff that is a good practice and is needed.

### LINKED ISSUE #11: Builds on MacOS
There is commented os in [ci.yml](https://github.com/coffee-and-rum/dealio/blob/6-remove-not-necessary/.github/workflows/ci.yml). We can try to run builds on it too 🛩️

### PR DESCRIPTION
**Pull request by LigasN**

### CODE REVIEW

**Comment by LigasN** [comment]
#11 is done in that issue already

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
-add_subdirectory(src)
+set(GIT_SHA
+    "Unknown"
+    CACHE STRING "SHA this build was generated from")
 
-# Don't even look at tests if we're not top level
-if(NOT PROJECT_IS_TOP_LEVEL)
-  return()
-endif()
+string(
```
Comment (Wneku1) [inline CMakeLists.txt:33]: what is this?

File: CMakePresets.json [hunk 2]
```diff
@@ -8,247 +8,104 @@
     "configurePresets": [
         {
             "name": "conf-common",
-            "description": "General settings that apply to all configurations",
+            "description": "general settings that apply to all configurations (targeted to release mode but ready to be overriden by debug settings)",
```
Comment (Wneku1) [inline CMakePresets.json:11]: looks better when it starts with a capital letter

**Review by Wneku1 — DISMISSED** [review]

**Comment by Wneku1** [comment]
cmake] CMake Warning:
[cmake]   Manually-specified variables were not used by the project:
[cmake] 
[cmake]     ENABLE_CLANG_TIDY
[cmake]     ENABLE_CPPCHECK

**Comment by LigasN** [comment]
> 

yeah, I left them in the presets, we will probably need them soon. Should I remove them for now?

**Comment by Ligas10105** [comment]
### The tests were performed on macOS Big Sur 11.6, and here are the results:

Clang:


GCC:

**Comment by LigasN** [comment]
> ### The tests were performed on macOS Big Sur 11.6, and here are the results:
> Clang: 
> 
> GCC: 

The GCC problem happens for Windows as well. I don't see a reason to spend more time on resolving this. Here is the [issue](https://github.com/coffee-and-rum/dealio/issues/14) to look at this problem in future.

**Comment by LigasN** [comment]
> > cmake] CMake Warning:
> > [cmake] Manually-specified variables were not used by the project:
> > [cmake]
> > [cmake] ENABLE_CLANG_TIDY
> > [cmake] ENABLE_CPPCHECK
> 
> yeah, I left them in the presets, we will probably need them soon. Should I remove them for now?

ok, I removed them, as well as msvc compiler option in presets that we are not testing in CI and probably no using

**Review by Wneku1 — APPROVED** [review]

**Pull request merged**